llvm/lib/Transforms/InstCombine/InstCombineInternal.h

```diff
@@ -585,6 +585,9 @@ class LLVM_LIBRARY_VISIBILITY InstCombinerImpl final
                                FPClassTest DemandedMask, KnownFPClass &Known,
                                unsigned Depth = 0);
 
+  /// Common transforms for add / disjoint or
+  Instruction *foldAddLike(BinaryOperator &I);
```

Comment:
If we add this kind of API, I think it would be better to make the signature `Value *LHS, Value *RHS, bool NUW, bool NSW`.